src/v/cloud_storage/tx_range_manifest.cc

```diff
@@ -0,0 +1,120 @@
+#include "cloud_storage/tx_range_manifest.h"
+
+#include "bytes/iobuf.h"
+#include "bytes/iobuf_istreambuf.h"
+#include "bytes/iobuf_ostreambuf.h"
+#include "cloud_storage/partition_manifest.h"
+#include "cloud_storage/types.h"
+#include "cluster/rm_stm.h"
+#include "json/istreamwrapper.h"
+#include "model/record.h"
+#include "utils/fragmented_vector.h"
+
+#include <rapidjson/document.h>
+#include <rapidjson/istreamwrapper.h>
+#include <rapidjson/ostreamwrapper.h>
+#include <rapidjson/rapidjson.h>
+#include <rapidjson/writer.h>
+
+namespace cloud_storage {
+
+remote_manifest_path generate_remote_tx_path(const remote_segment_path& path) {
+    return remote_manifest_path(fmt::format("{}.tx", path().native()));
+}
+
+tx_range_manifest::tx_range_manifest(
+  remote_segment_path spath,
+  const std::vector<cluster::rm_stm::tx_range>& range)
+  : _path(std::move(spath)) {
+    for (const auto& tx : range) {
+        _ranges.push_back(tx);
+    }
+    _ranges.shrink_to_fit();
+}
+
+tx_range_manifest::tx_range_manifest(remote_segment_path spath)
+  : _path(std::move(spath)) {}
+
+ss::future<> tx_range_manifest::update(ss::input_stream<char> is) {
+    using namespace rapidjson;
+    iobuf result;
+    auto os = make_iobuf_ref_output_stream(result);
+    co_await ss::copy(is, os);
+    iobuf_istreambuf ibuf(result);
+    std::istream stream(&ibuf);
+    Document m;
+    IStreamWrapper wrapper(stream);
+    m.ParseStream(wrapper);
+    update(m);
+}
+
+void tx_range_manifest::update(const rapidjson::Document& doc) {
+    _ranges = fragmented_vector<cluster::rm_stm::tx_range>();
+    auto version = doc["version"].GetInt();
+    auto compat_version = doc["compat_version"].GetInt();
```

Comment:
I get a bit lost with versions. Now data has both version and compat version and code has version and compat version. I expect code to only have current version and data to have current version (version of code that produced this data) and compat version (version of code that can read this data). 

And then we need to compare data's compat version with code's version.

Reply:
Code can have compat version too. E.g. if we remove support for some old format, compat version must be increased - then if data.version < code.compat_version, data is too old to be correctly parsed.

Here are the [serde comparisons](https://github.com/redpanda-data/redpanda/blob/35d346ae0beb031147c780a2a5135fd754348461/src/v/serde/serde.h#L413-L431) for reference.